decomposer.go

```diff
@@ -69,7 +71,19 @@ func (d *Decimal) Decompose(buf []byte) (form byte, negative bool, coefficient [
 	// Finite form.
 	negative = d.Negative
 	exponent = d.Exponent
-	coefficient = d.Coeff.Bytes()
+
+	sizeInBytes := (d.Coeff.BitLen() + 8 - 1) / 8 // math.Ceil(d.Coeff.BitLen()/8.0)
+	if cap(buf)-len(buf) >= sizeInBytes {
+		// it extends the buffer as the filling of bytes expects an already
+		// allocated slice.
+		buf = append(buf, make([]byte, sizeInBytes)...)
+
+		// We can fit the coefficient in the given buffer which prevents an
+		// allocation.
+		coefficient = d.Coeff.FillBytes(buf)
```

Comment:
`FillBytes` is only available in go1.15 and above. We were meaning to bump the minimum supported version of this library anyway, so that's ok. We'll just need to remove testing for go1.13 and go1.14 in `.github/workflows/go.yml`.

Reply:
At some point, we can also merge the conditional compilation `bigint_go1.X(_test).go` files away, but that's clean up which we can do separately.

Reply:
Thanks I completely forgot to check for version compatibility. I'm also adding 1.21 and 1.22 if that's ok.

Reply:
Is that needed? We did so in https://github.com/cockroachdb/apd/pull/131.

Reply:
Is that needed? We did so in https://github.com/cockroachdb/apd/pull/131.

Reply:
Very true, I'll remove the change here.